app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/api/MvvDateSerializer.kt

```diff
@@ -0,0 +1,31 @@
+package de.tum.`in`.tumcampusapp.component.ui.transportation.api
+
+import com.google.gson.*
+import org.joda.time.DateTime
+import java.lang.reflect.Type
+
+/**
+ * Parses the MVV specific dateTime, with json elements of:
+ * year, month, day, hour, minute
+ */
+class MvvDateSerializer : JsonDeserializer<DateTime> {
+
+    override fun deserialize(json: JsonElement?, typeOfT: Type?, context: JsonDeserializationContext?): DateTime {
+        if (json !is JsonObject) {
+            throw JsonParseException("Invalid MvvDateTime: $json")
```

Comment:
Does it make sense ton throw an `IOException` instead, @pfent? We can’t react to runtime exceptions with Retrofit, so this would result in a crash of the app. If we used an `IOException`, we could call `showErrorLayout()` in case of an exception.

Reply:
You could even make it extend `IOException` and use a custom exception message.

Reply:
You're both wrong. :dagger: 
And you should feel bad about it.

The `JsonParseException` is *literally* declared in Gson's `JsonDeserializer<T>` interface and the recommended way of error handling when deserializing JSON with Gson.

Gson's Javadoc states:
`@throws JsonParseException if json is not in the expected format of {@code typeofT}`

Reply:
@pfent How do those exceptions propagate through the system? As they’re not subclasses of IOException, Retrofit does not invoke the onFailure() callback when they occur.

Reply:
@thellmund I actually don't know.
Since Retrofit supports multiple deserialization mechanisms, I guess they handle erroneous parsing and handle that as a failure.
Alas our `DateSerializer` also throws a `JsonParseException` and failure handling there works fine.

Nevertheless, parsing data is still not in any way an I/O operation.